map.go

```diff
@@ -5,15 +5,19 @@ import (
 	"fmt"
 	"syscall"
 	"unsafe"
+
+	"github.com/koneu/natend"
```

Comment:
Importing a package for this is a lot in my opinion.
Something like this should work

```golang
import "encoding/binary"
var NativeEndian binary.ByteOrder

func init() {
	if isBigEndian() {
		NativeEndian = binary.BigEndian
	} else {
		NativeEndian = binary.LittleEndian
	}
}

func isBigEndian() (ret bool) {
	var i int = 0x1
	bs := (*[int(unsafe.Sizeof(0))]byte)(unsafe.Pointer(&i))
	if bs[0] == 0 {
		return true
	} else {
		return false
	}
}
```

[To quote Rob Pike](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=9m28s)
> A little copying is better than a little dependency.

Reply:
Good point, changed.